### ISSUE TITLE
Partial dates and duplicates handling in `derive_var_dthcaus()`

### ISSUE THREAD

**Issue by yangc71** [pharmaverse/admiral#822]
Issue: The observations are currently sorted by variable specified for `date` (expected as character variable) and sorted alphabetically. However, there is no imputation done to partial dates. So if multiple death records exist in different domains with partial dates, it is sorted like: "2020" is before "2020-12" and this is before "2020-12-06".

Action: Update the function to allow partial dates imputation.

If there is more than one observation with the same date within a source, a warning is issued (for example if there is more than one fatal AE). The warning can not be avoided and to select the desired observation the input dataset has to be sorted before it is passed to `derive_var_dthcaus()`. This should be changed such that it is possible to specify additional sort variables like AESEQ to avoid the warning.

**Comment by rossfarrugia** [comment]
thanks @tamarasenior for the efforts so far on this. once you're happy with the updates make a PR to merge into devel and then the PR checklist will pop up to talk you through the checks needed, e.g. maybe need to add something to the corresponding unit test program to cover the update. you could ask Annie to review then as she drafted the original issue.

**Comment by tamarasenior** [comment]
Hi @bundfussr, I have written the code for the second part of the requested update (handling duplicates), however I wanted to check before I push the updates that the date imputation should happen before the checking for duplicates? For example if the dates 2021-01 and 2021-01-01 are in the data and date imputation is set to FIRST, should this flag as a duplicate?

**Comment by rossfarrugia** [comment]
@tamarasenior i think thats still fine to flag as a duplicate. in this case the most logical thing in my mind would be to take the non-imputed date as occurring first, as that's likely the most reliable death cause as we have a full date entered by the site. Stefan will comment I'm sure when back and may correct me, but I just wanted to share my 2 cents in case useful whilst he's away.

**Comment by bundfussr** [comment]
@tamarasenior , I agree with @rossfarrugia .

**Comment by tamarasenior** [comment]
@bundfussr I want to check what the desired outcome is here - currently the derive_var_dthcaus() function is programmed so that if multiple records exist in different sources then the record is kept from the first source domain listed, regardless of how they are each is sorted. Do you want this functionality rewritten so that there is the option to either impute partial dates/select non-imputed dates in the instance of duplicate records across sources? Or do you just want the dthcaus_source function updated so that you can impute partial dates/select non-imputed dates within a single source?

**Comment by bundfussr** [comment]
I would update the `dthcaus_source()` function such that the imputation parameters like `date_imputation`, `time_imputation`, ... can be specified. In `derive_var_dthcaus()` the imputation should be done before the first/last observation is selected.

**Comment by tamarasenior** [comment]
@bundfussr thanks for the instruction, if you have time to review the updates that'd be great

**Comment by rossfarrugia** [comment]
@tamarasenior thanks for the efforts! if you're happy with the code you could make a pull request to devel as per https://pharmaverse.github.io/admiral/articles/development_process.html and add Stefan as reviewer once you run through all the suggested checks there.
